specs/rollup-node.md

```diff
@@ -59,31 +64,82 @@ currently only concerned with the specification of the rollup driver.
 
 [l2-chain-derivation]: #l2-chain-derivation
 
-This section specifies how the [rollup driver][g-rollup-driver] derives one L2 [deposit block][g-deposit-block] per
-every L1 block. The L2 block carries *[deposited transactions][g-deposited]* of two kinds:
+This section specifies how the [rollup driver][g-rollup-driver] derives a sequence of L2 blocks per sequencing window.
 
-- a single *[L1 attributes deposited transaction][g-l1-attr-deposit]* (always first)
-- zero or more *[user-deposited transactions][g-user-deposited]*
+Every L2 block carries transactions of two categories:
+
+- *[deposited transactions][g-deposited]*: two kinds:
+  - derived from the L1 chain: a single *[L1 attributes deposited transaction][g-l1-attr-deposit]* (always first).
+  - derived from [receipts][g-receipts]: zero or more *[user-deposited transactions][g-user-deposited]*.
+- *[sequenced transactions][g-sequencing]*: derived from [sequencer batches][g-sequencer-batch],
+  zero or more regular transactions, signed by L2 users.
 
 ------------------------------------------------------------------------------------------------------------------------
 
+## From L1 chain to Sequencing Window
+
+A [sequencing window][g-sequencing-window] is a fixed number consecutive L1 blocks that a derivation step takes as
+input. The window is identified by an `epoch`, equal to the block number of the first block in the window.
+
+As the full derivation of the L2 chain by the driver progresses each derivation step shifts the window forward by a
+single L1 block: the windows overlap.
+
+Each sequencing window is derived into a variable number of L2 blocks, depending on the timestamps of L1 and L2.
+
+The L2 has a fixed block time and no more than one batch per block,
+meaning that gaps between the batches (ordered by timestamp) are interpreted as batches with empty transaction-lists,
+thus construing L2 blocks that only contain deposit transaction(s).
+
+The L2 blocks produced by a sequencing window are bounded by timestamp:
+
+- `min_l2_timestamp = prev_l2_timestamp + l2_block_time`
+- `max_l2_timestamp = l1_timestamp + l2_block_time`, where `l1_timestamp` is the timestamp of the
+  first L1 block of the sequencing window. (maximum bound, may not be aligned with block time)
+
+If there are no batches present in the sequencing window then the L2 chain is extended up to `max_l2_timestamp` (incl.)
+with empty batches, but otherwise regular block derivation.
+
+Note that with short block times on L1 the L2 time may increment beyond the L1 time,
+but the longer target block time of L1 will correct back and allow the timestamps to align again.
+
```

Comment:
Imho this section is the weak part of the spec, I think we can better explain the relation between L1 and L2 blocks.

No particular suggestion though (any such suggestion would probably be equivalent to me trying to write an alternative explanation, which is probably better done in another PR.)